internal/signerverifier/ed25519/ed25519_test.go

```diff
@@ -0,0 +1,148 @@
+package ed25519
+
+import (
+	"context"
+	"encoding/hex"
+	"os"
+	"path"
+	"testing"
+
+	e "crypto/ed25519"
+
+	"github.com/adityasaky/gittuf/internal/signerverifier"
+	"github.com/adityasaky/gittuf/internal/tuf"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewSignerVerifierFromSecureSystemsLibFormat(t *testing.T) {
+	privateKeyPath := path.Join("test-data", "test-key")
+	privateKeyBytes, err := os.ReadFile(privateKeyPath)
+	if err != nil {
+		t.Error(err)
+	}
+
+	sv, err := NewSignerVerifierFromSecureSystemsLibFormat(privateKeyBytes)
+	if err != nil {
+		t.Error(err)
+	}
+
+	expectedPrivateString := "66f6ebad4aeb949b91c84c9cfd6ee351fc4fd544744bab6e30fb400ba13c6e9a3f586ce67329419fb0081bd995914e866a7205da463d593b3b490eab2b27fd3f"
+	expectedPrivate, err := hex.DecodeString(expectedPrivateString)
+	if err != nil {
+		t.Error(err)
+	}
+	expectedPrivateKey := e.PrivateKey(expectedPrivate)
+
+	expectedPublicString := "3f586ce67329419fb0081bd995914e866a7205da463d593b3b490eab2b27fd3f"
+	expectedPublic, err := hex.DecodeString(expectedPublicString)
+	if err != nil {
+		t.Error(err)
+	}
+	expectedPublicKey := e.PublicKey(expectedPublic)
+
+	assert.Equal(t, "52e3b8e73279d6ebdd62a5016e2725ff284f569665eb92ccb145d83817a02997", sv.keyID)
+	assert.Equal(t, expectedPrivateKey, sv.private)
+	assert.Equal(t, expectedPublicKey, sv.public)
+
+	publicKeyPath := path.Join("test-data", "test-key.pub")
+	publicKeyBytes, err := os.ReadFile(publicKeyPath)
+	if err != nil {
+		t.Error(err)
+	}
+
+	sv, err = NewSignerVerifierFromSecureSystemsLibFormat(publicKeyBytes)
```

Comment:
I'm assuming these are 2 logically different tests because we're reading in new data.

It might be helpful to break this up using t.Run() to make this distinction clearer. 2 ways you could do this:

1. just wrap each case with t.Run individually. e.g.
```go
t.Run("foo", func(t *testing.T) {
...
})
t.Run("bar", func(t *testing.T) {
...
})
```
2. Set up a table driven test.

No strong preferences here. Up to you!